### DISCUSSION TITLE
QDate range localization

### DISCUSSION THREAD

**Discussion by albertoctopgraf** [quasarframework/quasar#10439] [General - Components / Directives / etc] [answered]
I couldn't localize text "days" in QDate with range directive.

Without custom localization
![image](https://user-images.githubusercontent.com/89253223/130223320-8ebb5649-43ba-4ad5-aefd-e87ce2b52d12.png)

With custom localiztion (italian)
![image](https://user-images.githubusercontent.com/89253223/130223529-bcff10d3-35f9-418f-b27a-ff29bcfda335.png)

I follow the documentation (https://quasar.dev/vue-components/date#custom-ad-hoc-locale)

- Template
![image](https://user-images.githubusercontent.com/89253223/130224310-203e2f05-774b-4576-8b71-64b8da7670fb.png)

- Script
setup(props, context) {
    ....
    const period = ref<DateRange | string>({
      from: "2021/08/18",
      to: "2021/08/20",
    });
    const dateLocale = computed<any>(() => {
      return {
        days: t("datePicker.days").split("_"),
        daysShort: t("datePicker.daysShort").split("_"),
        months: t("datePicker.months").split("_"),
        monthsShort: t("datePicker.monthsShort").split("_"),
        firstDayOfWeek: getFirstDayOfWeek(),
      };
    });
    ....
    return {
      period,
      dateLocale,
      ....
    };
}

Could you help me?

Thanks.

Alberto

**Comment by ontwikkelfabriek** [comment]
@albertoctopgraf 

> https://quasar.dev/vue-components/date#custom-ad-hoc-locale

This example shows how you can change the locale for just this component( at runtime). More often you want your entire Quasar app to be in a specific language. 

So is this really what you want? Just for the Q-Date to be displayed in **Italian** and the rest of the app in default **English**? Or do you want the **entire** app to be displayed in Italian?

**Reply by albertoctopgraf** [reply]
Thank you.
The app is multilanguage. I need only to display the Q-Date in the app languages.
I check the quasar global locale, it's "it-IT".

**Comment by ontwikkelfabriek** [comment]
@albertoctopgraf 

eum ok, but if the global locale is set to `it-IT` then why is the Q-date not displaying `Italian` by `default`? The whole app should be in Italian.

**Comment by albertoctopgraf** [comment, marked as answer]
I solve the issue removing the custom localization on the QDate and setting in setup the Quasar package language:

void import(
  /* webpackInclude: /(it|fr|en-GB)\.js$/ */
  "quasar/lang/" + appLocale
).then((lang) => {
  $q.lang.set(lang.default);
});

Thanks for the support.

Bye
